rocminfo.cc

```diff
@@ -50,10 +55,25 @@
 #include "hsa/hsa.h"
 #include "hsa/hsa_ext_amd.h"
 
+#define RET_IF_HSA_INIT_ERR(err) {                      \
+  if ((err) != HSA_STATUS_SUCCESS) {                    \
+    CheckInitError(err);                                \
+    RET_IF_HSA_ERR(err);                                \
+  }                                                     \
+}
+
 #define RET_IF_HSA_ERR(err) { \
   if ((err) != HSA_STATUS_SUCCESS) { \
-    printf("hsa api call failure at line %d, file: %s. Call returned %d\n", \
-                                                   __LINE__, __FILE__, err); \
+    char err_val[8];                                            \
```

Comment:
I noticed the err_val is not null-terminated. How about avoid using a char array with something like this:

 #define RET_IF_HSA_ERR(err) { \
   if ((err) != HSA_STATUS_SUCCESS) { \
    printf("hsa api call failure at line %d, file: %s. Call returned 0x%X\n", \
                                                    __LINE__, __FILE__, err); \
    const char *status_str; \
    hsa_status_t err2; \
    err2 = hsa_status_string(err,&status_str); \
    if (err2 != HSA_STATUS_SUCCESS) { \
      printf("hsa_status_string return 0x%X for error code 0x%X.\n", err2, err); \
    } else { \
      printf("%s\n", status_str); \
    } \
    return (err); \
   } \
 }